File: tests/fixtures/comment-extraction/sample.py
# Single line comment before function

def add(a: int, b: int) -> int:
    """
    Docstring for add function

    Args:
        a: First number
        b: Second number

    Returns:
        Sum of a and b
    """
    # Inline comment
    return a + b  # End of line comment


class User:
    """
    Multi-line docstring
    describing the User class
    """

    def __init__(self, name: str):
        """
        Constructor docstring

        Args:
            name: User name
        """
        self.name = name

    # TODO: Implement user validation
    def validate(self) -> bool:
        # FIXME: This is a placeholder implementation
        return len(self.name) > 0

    # NOTE: This method should be optimized
    # HACK: Quick fix for performance issue
    def process(self) -> None:
        """Process user data"""
        pass


async def fetch_data() -> None:
    """
    Async function with docstring

    Raises:
        Exception: When network request fails
    """
    # XXX: Temporary workaround
    # BUG: Known issue with error handling
    pass


# Global variable with comment
MAX_RETRIES = 3  # Maximum retry attempts

File: tests/fixtures/comment-extraction/sample.rs
// Single line comment before function

/// Doc comment for add function
///
/// # Arguments
/// * `a` - First number
/// * `b` - Second number
///
/// # Returns
/// Sum of a and b
pub fn add(a: i32, b: i32) -> i32 {
    // Inline comment
    a + b // End of line comment
}

/*
 * Multi-line block comment
 * describing the User struct
 */
pub struct User {
    /// Property comment
    pub name: String,
}

impl User {
    /// Constructor comment
    ///
    /// # Arguments
    /// * `name` - User name
    pub fn new(name: String) -> Self {
        User { name }
    }

    // TODO: Implement user validation
    pub fn validate(&self) -> bool {
        // FIXME: This is a placeholder implementation
        !self.name.is_empty()
    }

    // NOTE: This method should be optimized
    // HACK: Quick fix for performance issue
    pub fn process(&self) {
        /* Implementation here */
    }
}

/// Async function with doc comment
///
/// # Errors
/// Returns error when network request fails
pub async fn fetch_data() -> Result<(), Box<dyn std::error::Error>> {
    // XXX: Temporary workaround
    // BUG: Known issue with error handling
    Ok(())
}

/// Maximum retry attempts
pub const MAX_RETRIES: u32 = 3;
